flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FileStoreSource.java

```diff
@@ -130,4 +119,42 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
             return new StaticFileStoreSplitEnumerator(context, snapshot, splits);
         }
     }
+
+    private DataFilePlan batchScanSplits(DataTableScan scan) {
+        return scan.plan();
+    }
+
+    private DataFilePlan continuousFirstScanSplits(
+            DataTableScan scan, SnapshotManager snapshotManager) {
+        switch (startupMode) {
+            case FULL:
+                DataFilePlan plan;
+                if (table.options().changelogProducer()
+                        == CoreOptions.ChangelogProducer.FULL_COMPACTION) {
+                    // Read the results of the last full compaction.
+                    // Only full compaction results will appear on the max level.
+                    plan = scan.withLevel(table.options().numLevels() - 1).plan();
+                } else {
+                    plan = scan.plan();
+                }
+                return plan;
+            case LATEST:
+                return new DataFilePlan(
+                        snapshotManager.latestSnapshotId(), Collections.emptyList());
+            case FROM_TIMESTAMP:
```

Comment:
Could the `log.scan` and `log.scan.timestamp-millis` merge to `log.scan`? If the `log.scan` is configured to certain timestamp, this means the `startUpMode` is `FROM_TIMESTAMP`, which solution is referred to `read.start-commit` in Hudi.

Reply:
Good point, I think we can improve this later, because now we can not unset option in Flink SQL, we can wait for that.

Reply:
Maybe we can let the default value of `log.scan` is none, and explain in comments?

Reply:
I created https://issues.apache.org/jira/browse/FLINK-30110 for this.